cmake/winml.cmake

```diff
@@ -196,6 +200,10 @@ add_dependencies(winml_lib_core winml_api_native_internal)
 target_link_libraries(winml_lib_core PRIVATE wil)
 target_link_libraries(winml_lib_core PRIVATE onnxruntime_providers_dml)
 
+# add it to the onnxruntime shared library
+set(onnxruntime_winml windowsapp.lib -WHOLEARCHIVE:$<TARGET_FILE:winml_lib_core>)
+list(APPEND onnxruntime_EXTERNAL_DEPENDENCIES winml_lib_core)
+
```

Comment:
is this going to live here in the final version? seems odd to add things to a different project

Reply:
that was the idea.    onnxruntime_EXTERNAL_DEPENDENCIES is a property setup just for this purpose.  for other cmake files to add things to the the final shared_lib

---
In reply to: [347039511](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347039511) [](ancestors = 347039511)